### PR TITLE
test: add coverage for DeepLinkViewModel error states

### PR DESCRIPTION
**Pull request by nonproto**
💡 What: Added `DeepLinkViewModelTest.kt` with a unit test targeting the "sad path" in `handleDeepLink`. Fixed testpilot.md dates. Re-initialized `Injekt = KotlinInjektRegistrar()` in `@Before` block to prevent `DoubleRegistrationException`.
🎯 Why: To improve test coverage by validating edge cases. Specifically, ensuring that missing UUID mappings correctly result in a `DeepLinkState.Error` with the appropriate error message, protecting against regressions in the app's deep link routing logic. Re-initializing `Injekt` ensures the test suite doesn't fail when multiple testing files interact with the singleton `Injekt` instance.

---
*PR created automatically by Jules for task [8571812100257383768](https://jules.google.com/task/8571812100257383768) started by @nonproto*

### CODE REVIEW

File: app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.nekomanga.data.database.repository.MangaRepository
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.addSingleton
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt:22]: > medium
> 
> Import `KotlinInjektRegistrar` to support re-initializing the global `Injekt` instance in the setup block.

@jules

File: app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        Dispatchers.resetMain()
+        val fields = Injekt::class.java.declaredFields
+        for (field in fields) {
+            if (field.name == "registrars") {
+                field.isAccessible = true
+                val map = field.get(Injekt) as MutableMap<*, *>
+                map.clear()
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (nonproto) [reply app/src/test/java/org/nekomanga/presentation/screens/deepLink/DeepLinkViewModelTest.kt:65]: > medium
> 
> Instead of using reflection to clear the internal `registrars` map in `tearDown`, re-initialize the global `Injekt` variable with a new `KotlinInjektRegistrar()` at the start of each test setup. This is cleaner, safer, and avoids fragile reflection logic.
> 
> ```kotlin
>     @Before
>     fun setup() {
>         Injekt = KotlinInjektRegistrar()
>         Dispatchers.setMain(testDispatcher)
> 
>         mockMangaMappings = mockk()
>         mockSourceManager = mockk()
>         mockMangaDex = mockk()
>         mockMangaRepository = mockk()
> 
>         every { mockSourceManager.mangaDex } returns mockMangaDex
> 
>         Injekt.addSingleton(mockMangaMappings)
>         Injekt.addSingleton(mockSourceManager)
>         Injekt.addSingleton(mockMangaRepository)
> 
>         viewModel = DeepLinkViewModel()
>     }
> 
>     @After
>     fun tearDown() {
>         Dispatchers.resetMain()
>     }
> ```
> 
> References

@jules
